tools/pylib/boutdata/squashoutput.py

```diff
@@ -0,0 +1,150 @@
+    """
+    Collect all data from BOUT.dmp.* files and create a single output file.
+
+    Parameters
+    ----------
+    datadir : str
+        Directory where dump files are and where output file will be created.
+        default "."
+    outputname : str
+        Name of the output file. File suffix specifies whether to use NetCDF or
+        HDF5 (see boututils.datafile.DataFile for suffixes).
+        default "BOUT.dmp.nc"
+    format : str
+        format argument passed to DataFile
+        default "NETCDF4"
+    tslice : [int, int, int]
+        lower, upper, stride values to slice the t-dimension
+        default [None, None, None]
+    xslice : [int, int, int]
+        lower, upper, stride values to slice the x-dimension
+        default [None, None, None]
+    yslice : [int, int, int]
+        lower, upper, stride values to slice the y-dimension
+        default [None, None, None]
+    zslice : [int, int, int]
+        lower, upper, stride values to slice the z-dimension
+        default [None, None, None]
+    singleprecision : bool
+        If true convert data to single-precision floats
+        default False
+    """
+
+    fullpath = os.path.join(datadir,outputname)
+    if os.path.isfile(fullpath):
+        raise ValueError(fullpath+" already exists. Collect may try to read from this file, which is presumably not desired behaviour.")
+
+    # Check *slice arguments, and make sure length is at least 3
+    if len(tslice) < 3:
+        tslice += [None]*(3-len(tslice))
+    elif len(tslice) > 3:
+        raise ValueError("Can provide at most 3 arguments for tslice")
+    if len(xslice) < 3:
+        xslice += [None]*(3-len(xslice))
+    elif len(xslice) > 3:
+        raise ValueError("Can provide at most 3 arguments for xslice")
+    if len(yslice) < 3:
+        yslice += [None]*(3-len(yslice))
+    elif len(yslice) > 3:
+        raise ValueError("Can provide at most 3 arguments for yslice")
+    if len(zslice) < 3:
+        zslice += [None]*(3-len(zslice))
+    elif len(zslice) > 3:
+        raise ValueError("Can provide at most 3 arguments for zslice")
+
+    # useful object from BOUT pylib to access output data
+    outputs = BoutOutputs(datadir, info=False, xguards=True, yguards=True)
+    outputvars = outputs.keys()
+    t_array_index = outputvars.index("t_array")
+    outputvars.append(outputvars.pop(t_array_index))
+
+    # Create single file for output and write data
+    with DataFile(fullpath,create=True,write=True,format=format) as f:
+        for varname in outputvars:
+            print(varname)
+
+            var = outputs[varname]
+            if singleprecision:
+                var = BoutArray(numpy.float32(var), var.attributes)
+            if var.attributes["bout_type"] == "Field3D_t":
+                var = var[tslice[0]:tslice[1]:tslice[2], xslice[0]:xslice[1]:xslice[2], yslice[0]:yslice[1]:yslice[2], zslice[0]:zslice[1]:zslice[2]]
```

Comment:
I think this could be tidied up if you used `slice` objects instead of lists/tuples here. Possibly even take them as arguments in the first place? You could check if you got a `list`/`tuple` with `isinstance(tslice, (list, tuple))` and then just convert to `slice`